olm/open-liberty-0.0.1.clusterserviceversion.yaml

```diff
@@ -190,19 +190,38 @@ spec:
       kind: OpenLiberty
       name: openliberties.openliberty.io
       resources:
-      - kind: Service
+      - kind: ConfigMap
         name: ""
         version: v1
-      - kind: StatefulSet
+      - kind: Deployment
+        name: ""
+        version: v1
```

Comment:
Are these version numbers important?  I know when we added the roles and role bindings to the CSV, some of the version numbers were different (ie v1alpha1 vs v1beta1 vs v1)

Reply:
They are. That is the version of the resources that are being used and owned by the CRD.
The doc on this is very sparse, but from the code, the test seems gather resource information through a proxy deployment that sees the resources (kind and version) being used by the CRD. It then compares that information to what it is specified in the CSV->customresourcedefinitions.resources. If it doesn't find a version/kind match, it displays what is missing (kind/version).

As for the version, the tooling seems to take information from the yaml files that make the operator. For example:

kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1

apiVersion: rbac.authorization.k8s.io/v1
kind: Role

apiVersion: v1
kind: ServiceAccount

apiVersion: apps/v1
kind: Deployment

etc.

For those resources that were already in the CSV such Pod or ConfigMap, etc. it is not evident where the version is coming from, but the test tooling seems to be figuring it out.